### PR TITLE
Rdtk implementation

### PR DESCRIPTION
**Pull request by SamiFdev**
toolkit seems to be working fine, just having the issue with it doing the second api request.

### CODE REVIEW

**Comment by SamiFdev** [comment]
Added a slice for single coin search. Current issue seems to be getting the "id" back from the search. Looked into it and I believe it's passed correctly inside the asyncThunk. "id" just keeps coming back as undefined. Spent a few hours trying to trouble shoot this which leads me to think it's just a logical thing I am overlooking atm. 

In main content component I have the "searched coin" setting and console logging (just realized its commented out but still is working). So the state is updating to read my searched coin in the input field. I think its all just broken because of my asyncThunk function inside of singleCoinSlice.js. 

In plain english, what I think is happening in main content, is I am setting a searched coin and firing off the fetchSingleCoin function trying to pass in the searched coin but must be doing it incorrectly. I toyed around with a lot all afternoon and this is what I ended up with after undoing and redoing for awhile after research. If you find the solution i'd love comments on what It is so I can correct it tomorrow Thanks.

File: src/components/MainContent/index.js [hunk 1]
```diff
@@ -1,41 +1,38 @@
...
+import { useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
 import { fetchSingleCoin } from "../../features/singleCoinSlice";
+import Loader from "../Loader";
+import Error from "../ErrorAlert";
 
 function MainContent() {
-    const [searchedCoin, setsearhedCoin] = useState("");
-    const [wasDataFetched, setWasDataFetched] = useState(false);
+    const [searchedCoin, setSearhedCoin] = useState("");
```
Comment (arfusop) [inline src/components/MainContent/index.js]: Typo...
`setSearhedCoin` -> `setSearchedCoin`

File: src/components/MainContent/index.js [hunk 2]
```diff
@@ -1,41 +1,38 @@
...
-        </div>
+        <section>
+            {loading ? <Loader /> : null}
+            {error ? <Error /> : null}
+            <div className={styles.searchBarContainer}>
+                <input
+                    className={styles.searchBar}
+                    type="text"
+                    value={searchedCoin}
+                    onChange={(e) => setSearhedCoin(e.target.value)}
```
Comment (arfusop) [inline src/components/MainContent/index.js]: Move this out of the onChange into its own function.

```
const handleInputChange = e =>
...
<input ... onChange={handleInputChange}
```

File: src/features/singleCoinSlice.js [hunk 3]
```diff
@@ -6,7 +6,6 @@ export const fetchSingleCoin = createAsyncThunk(
         const data = await fetch(
             `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&ids=${id}`
         ).then((res) => res.json());
-        //coming back as undefined
         console.log("test", id);
```
Comment (arfusop) [inline src/features/singleCoinSlice.js]: Delete the log

File: src/index.js [hunk 4]
```diff
@@ -1,21 +1,15 @@
-import React from 'react';
-import { createRoot } from 'react-dom/client';
-import { Provider } from 'react-redux';
-import { store } from './app/store';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
-import './index.css';
+import React from "react";
+import { createRoot } from "react-dom/client";
+// import { Provider } from "react-redux";
```
Comment (arfusop) [inline src/index.js]: Delete comments

**Review by arfusop — CHANGES_REQUESTED** [review]
Small changes

File: src/components/MainContent/index.js [hunk 5]
```diff
@@ -1,27 +1,42 @@
...
-                type="text"
-                value={searchedCoin}
-                onChange={(e) => setsearhedCoin(e.target.value)}
-                placeholder="Search a coin"
-            />
-            <button onClick={handleSubmit} className={styles.submitButton}>
-                Submit
-            </button>
-        </div>
+        <section>
```
Comment (arfusop) [inline src/components/MainContent/index.js]: Noticed this now, you added a section without a class that just wraps your div.searchBarContainer

Why have a double container? Either add the class to the section and delete the div (which is what I'd do) or remove the section. Having both is unneeded semantically, and structurally.

File: src/utils/api.js [hunk 6]
```diff
@@ -1,26 +1,26 @@
-import axios from "axios";
+// import axios from "axios";
```
Comment (arfusop) [inline src/utils/api.js]: Remove comments

**Review by arfusop — CHANGES_REQUESTED** [review]

**Review by arfusop — APPROVED** [review]
On pull requests, mark each comment as resolved if you've addressed and made a fix for them. Every comment i've left is technically left open on your PR. Helps everyone involved in the PR if you mark as resolved everything you hit on when you re-request for a fresh review.
